### PR TITLE
add options for nondivergent and divergent cmap

### LINKED ISSUE #2394: Change default colormaps
#### Problem description

xarrays plotting module is awesome because it detects automatically if the data is divergent or not and adjusts the colormap accordingly.

But what if I do not particularly like the default colormap choice for either data type (e.g. my current boss is not a fan of matplotlibs [viridis](https://matplotlib.org/examples/color/colormaps_reference.html))?

@shoyer mentioned on [stackexchange](https://stackoverflow.com/questions/52114364/change-the-default-colorbar-in-xarray-for-non-diverging-data?noredirect=1#comment91188566_52114364) that this is not currently possible. 

I would be happy to submit a PR, if I can get some guidance on where to implement this.

What does everyone think about a syntax this shoule be implemented with?

Something like:

```
with xr.set_options(diverging_cmap='RdYlBl'):
    ...
with xr.set_options(nondiverging_cmap='magma'):
    ....
```
Or is there a better way to do this?
I would personally prefer to be able to set this at the beginning of a notebook, so that it is applied to all following plots. Maybe something like the `mpl.rcparams` is a way to go here?

### PR DESCRIPTION
**Pull request by jbusecke**
- [ ] Closes #2394 (remove if there is no corresponding issue, which should only be the case for minor changes)
 - [ ] Tests passed (for all non-documentation changes)
- [ ] Tests added (for all bug fixes or enhancements)
  - [ ] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API (remove if this change should not be visible to users, e.g., if it is an internal clean-up, or if this is part of a larger project that will be documented later)

### CODE REVIEW

**Comment by dcherian** [comment]
Should we call it `cmap_sequential` instead of just `cmap`?

File: xarray/tests/test_plot.py [hunk 1]
```diff
@@ -470,6 +471,16 @@ def test_center(self):
         assert cmap_params['levels'] is None
         assert cmap_params['norm'] is None
 
+    def test_cmap_option(self):
```
Comment (dcherian) [inline xarray/tests/test_plot.py]: `test_cmap_sequential_option`.

Nice test btw.

File: xarray/tests/test_plot.py [hunk 2]
```diff
@@ -470,6 +471,16 @@ def test_center(self):
...
+        with xr.set_options(cmap_sequential='magma'):
+            cmap_params = _determine_cmap_params(self.data)
+            assert cmap_params['cmap'] == 'magma'
+
+    def test_cmap_divergent_option(self):
+        with xr.set_options(cmap_divergent='magma'):
```
Comment (dcherian) [inline xarray/tests/test_plot.py]: can you add a test where you set it to `mpl.cm.magma`. It should all just work but would be nice to explicitly test for that option.

File: xarray/tests/test_plot.py (hunk 1 above)
Comment (jbusecke) [reply xarray/tests/test_plot.py]: Thanks!

File: xarray/tests/test_plot.py (hunk 2 above)
Comment (jbusecke) [reply xarray/tests/test_plot.py]: Just added an explicit test.

**Comment by jbusecke** [comment]
Awesome. Thanks a lot for the feedback @dcherian.

**Comment by jbusecke** [comment]
Yikes. Also sorry for this merge fail commit. I am sitting in a conference :-).

File: xarray/tests/test_plot.py [hunk 3]
```diff
@@ -470,6 +471,21 @@ def test_center(self):
...
 
+    def test_cmap_sequential_option(self):
+        with xr.set_options(cmap_sequential='magma'):
+            cmap_params = _determine_cmap_params(self.data)
+            assert cmap_params['cmap'] == 'magma'
+
+    def test_cmap_sequential_explicit_option(self):
+        with xr.set_options(cmap_sequential=mpl.cm.magma):
+            cmap_params = _determine_cmap_params(self.data)
+            assert cmap_params['cmap'] == 'magma'
```
Comment (dcherian) [inline xarray/tests/test_plot.py]: `== mpl.cm.magma`?

**Comment by jbusecke** [comment]
I am unsure if that failure is due to a time out or changes in the PR. Is ther anything else that I should change before merge? Again, many thanks for the help in getting this forward.

**Comment by shoyer** [comment]
I restarted the timed out test, but it does look non-related. Merging -- thanks!

**Pull request merged**